### ISSUE TITLE
iOS Bundle Resource targets with auto generated plists may fail upload to App Store

### ISSUE THREAD

**Issue by kwridan** [tuist/tuist#1360]
**Describe the bug**
In the a resource bundle target (product: `.bundle`) is used with an auto generated plist `infoPlist: .default`, the generated project can produce a `.bundle` that can fail upload validation.

```
Error 1: ITMS-90535 - Unexpected CFBundleExecutable Key
```

**To Reproduce**

1. Go to `fixtures/ios_app_with_framework_and_resources/StaticFramework`
2. Update the manifest to auto generate the plist for the bundle target

```swift
import ProjectDescription

let project = Project(name: "StaticFramework",
                      targets: [
                          Target(name: "StaticFramework",
                                 platform: .iOS,
                                 product: .staticFramework,
                                 bundleId: "io.tuist.StaticFramework",
                                 infoPlist: "Config/StaticFramework-Info.plist",
                                 sources: "Sources/**",
                                 dependencies: []),
                          Target(name: "StaticFrameworkResources",
                                 platform: .iOS,
                                 product: .bundle,
                                 bundleId: "io.tuist.StaticFrameworkResources",
                                 infoPlist: .default, // <---
                                 sources: [],
                                 resources: "Resources/**",
                                 dependencies: []),
])
```
3. Run `tuist generate`
4. Inspect the generated plist file

it contains:
```
	<key>CFBundleExecutable</key>
	<string>$(EXECUTABLE_NAME)</string>
```

5. The generated target has an empty **Compile Sources** build phase

**Expected behavior**
- The generated plist for iOS bundles should omit that key. 
- The **Compile Sources** build phase should be omitted seeing its empty.

**Desktop (please complete the following information):**

- Tuist 1.8.0

**Additional context**

@remover mentioned on slack the generated project when built produced a binary for the resource target, this can also cause upload rejections:

e.g.

```
Error 1: ITMS-90171 - Invalid Bundle Structure
Invalid Bundle Structure - The binary file ‘MyApp.app/FrameworkResources.bundle/FrameworkResources’ is not permitted.
``` 

@remover was able to resolve this by removing the empty **compile sources** build phase. Oddly I wasn't able to see a binary when testing in a fixture or in any of our production targets, I wonder if something else (like a build setting) could cause the build process to spit out a binary for the bundle target.

**Update:**

While troubleshooting this we noticed that the generated project doesn't explicitly set a `MACH_O_TYPE` setting, leaving it to be inherited from xcconfig files

when we compared the fixture against the target with the binary we noticed:

```
• MACH_O_TYPE
    ◦ mh_bundle
    ◦ staticlib
```

**Update 2:**

Turns out, having the `VERSIONING_SYSTEM` setting set in the resource bundle with an empty **Compile Sources** phase will generate a binary in the final product. Thanks to @remover troubleshooting this.

**Comment by EduardLev** [comment]
I see this issue was reopened, @pepibumur were you able to resolve this?

**Comment by remover** [comment]
FWIW "Update 2" is the actual problem. This issue may have been reopened since we recently switched to the native tuist way of specifying resource bundles. As a result of doing this an old build setting (`VERSIONING_SYSTEM `) took effect since we were no longer overriding it. cc @RomainBoulay

**Comment by kwridan** [comment]
I believe we can close this issue it has been addressed and additionally #2376 ensures we no longer add the sources build phase to targets that don't support it (like iOS .bundle targets).
